src/main/java/cubicchunks/world/column/CubeMap.java

```diff
@@ -208,6 +204,6 @@ private int binarySearch(int cubeY) {
 			}
 		}
 
-		return mid; // not found :(
+		return start; // not found :(
```

Comment:
There is Collections.binarySearch. It returns negative value when element isn't found (-insertionPoint - 1). Seems like good thing.